remotes/docker/resolver.go

```diff
@@ -374,11 +378,15 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 		}
 	}
 
-	if lastErr == nil {
-		lastErr = errors.Wrap(errdefs.ErrNotFound, ref)
+	// If above loop terminates without return, then there was an error.
+	// "firstErr" contains the first non-404 error. That is, "firstErr == nil"
+	// means that either no registries were given or each registry returned 404.
```

Comment:
would it make sense to separate not found due to 404 from all given registries and no registries were given here? Maybe also build a list of 404 paths? or do you think the log is sufficient?

Reply:
nvm I addressed with additional log..